catchup/service.go

```diff
@@ -168,6 +168,12 @@ func (s *Service) fetchAndWrite(fetcher rpcs.Fetcher, r basics.Round, prevFetchC
 		default:
 		}
 
+		// Stop retrying after a while.
+		if i > 20 {
+			s.log.Errorf("fetchAndWrite(%v): failed to fetch block many times", )
+			return false
+		}
+
```

Comment:
I don't think that we want that, as it competes with the existing catchup logic.

Reply:
Is there a number between 20 and 6 million that you think would be alright?
```
fetchAndWrite(3317176): block contents do not match header (attempt 6242741)
```

Reply:
yes; we want to have a number that is greater than the number of relays so that every relay is likely to be attempted before giving up.